lib/spack/spack/binary_distribution.py

```diff
@@ -0,0 +1,481 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+import os
+import re
+import tarfile
+import yaml
+import shutil
+
+import llnl.util.tty as tty
+from spack.util.gpg import Gpg
+from llnl.util.filesystem import mkdirp, join_path, install_tree
+from spack.util.web import spider
+import spack.cmd
+import spack
+from spack.stage import Stage
+import spack.fetch_strategy as fs
+import spack.relocate
+from contextlib import closing
+import spack.util.gpg as gpg_util
+import hashlib
+
+
+def has_gnupg2():
+    try:
+        gpg_util.Gpg.gpg()('--version', output=os.devnull)
+        return True
+    except Exception:
+        return False
+
+
+def buildinfo_file_name(prefix):
+    """
+    Filename of the binary package meta-data file
+    """
+    name = prefix + "/.spack/binary_distribution"
+    return name
+
+
+def read_buildinfo_file(prefix):
+    """
+    Read buildinfo file
+    """
+    filename = buildinfo_file_name(prefix)
+    with open(filename, 'r') as inputfile:
+        content = inputfile.read()
+        buildinfo = yaml.load(content)
+    return buildinfo
+
+
+def write_buildinfo_file(prefix):
+    """
+    Create a cache file containing information
+    required for the relocation
+    """
+    text_to_relocate = []
+    binary_to_relocate = []
+    blacklist = (".spack", "man")
+    for root, dirs, files in os.walk(prefix, topdown=True):
```

Comment:
If I understand correctly, the below calls to ```relocate.get_filetype``` take a while? That makes sense. It would be great if you could add a comment on that. Even if this stays the same because you are short on time and for no other reason that works too - I don't consider it a problem just a preference.